runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpClientFactory.java

```diff
@@ -868,962 +1027,3075 @@ int decode(
+                    cleanupNetwork(traceId, authorization);
+                }
+                else if (HttpState.replyClosing(state))
+                {
+                    state = HttpState.closeReply(state);
+
+                    pool.exchanges.forEach((id, exchange) -> exchange.cleanup(traceId, authorization));
+
+                    doNetworkEnd(traceId, authorization);
+                }
+            }
+
+            if (exchange != null && !HttpState.replyClosed(exchange.state))
+            {
+                doNetworkWindow(traceId, budgetId, exchange.responsePad, decodeSlotReserved);
+            }
+        }
+
+        private void onDecodeHttp1HeadersError(
+            long traceId,
+            long authorization)
+        {
+            cleanupNetwork(traceId, authorization);
+        }
+
+        private void onDecodeHttp1BodyError(
+            long traceId,
+            long authorization)
+        {
+            cleanupNetwork(traceId, authorization);
+        }
+
+        private void onDecodeHttp1Headers(
+            long traceId,
+            long authorization,
+            HttpBeginExFW beginEx)
+        {
+            exchange.doResponseBegin(traceId, authorization, beginEx);
+
+            final HttpHeaderFW connection = beginEx.headers().matchFirst(h -> HEADER_CONNECTION.equals(h.name()));
+            if (connection != null && connectionClose.reset(connection.value().asString()).matches())
+            {
+                exchange.state = HttpState.closingReply(exchange.state);
+            }
+
+            final HttpHeaderFW status = beginEx.headers().matchFirst(h -> HEADER_STATUS.equals(h.name()));
+            if (status != null &&
+                encoder == HttpEncoder.HTTP_1_1 &&
+                STATUS_101.equals(status.value()))
+            {
+                pool.onUpgradedOrClosed(this);
+            }
+        }
+
+        private void onDecodeHttp1HeadersOnly(
+            long traceId,
+            long authorization,
+            Flyweight extension)
+        {
+            exchange.doResponseEnd(traceId, authorization, extension);
+        }
+
+        private int onDecodeBody(
+            long traceId,
+            long authorization,
+            long budgetId,
+            DirectBuffer buffer,
+            int offset,
+            int limit,
+            Flyweight extension)
+        {
+            return exchange.doResponseData(traceId, authorization, buffer, offset, limit, extension);
+        }
+
+        private void onDecodeHttp1Trailers(
+            long traceId,
+            long authorization,
+            Flyweight extension)
+        {
+            exchange.doResponseEnd(traceId, authorization, extension);
+        }
+
+        private void doEncodeH2cHeaders(
+            long traceId,
+            long authorization,
+            long budgetId,
+            Array32FW<HttpHeaderFW> headers,
+            Map<String8FW, String16FW> overrides)
+        {
+            headersMap.clear();
```

Comment:
Let's just create a new `LinkedHashMap<String, String>` here instead of hoisting the reusable map, as we haven't yet optimized the codepath for regular requests and the `h2c` path is significantly less frequent.